services:
  db:
    image: postgres:latest
    restart: on-failure
    env_file:
      - .env
    expose:
      - "5432"
    volumes:
      - pg_data:/var/lib/postgresql/data
    healthcheck:
      test: [ "CMD-SHELL", "-c", "pg_isready -U $POSTGRES_USER" ]
      interval: 10s
      timeout: 5s
      retries: 5
    networks:
      - memont

  memont_diary:
    build: .
    tty: true
    ports:
      - "8000:8000"
    command: sh -c "python manage.py migrate && python manage.py runserver 0.0.0.0:8000"
    volumes:
      - .:/app
    env_file:
      - .env
    depends_on:
      db:
        condition: service_healthy
    networks:
      - memont

  redis:
    image: redis:latest
    restart: on-failure
    expose:
      - "6379"
    networks:
      - memont

volumes:
  pg_data:

networks:
  memont:
    driver: bridge